src/ngraph/runtime/gpu/gpu_external_function.cpp

```diff
@@ -241,7 +242,18 @@ runtime::gpu::GPU_ExternalFunction::GPU_ExternalFunction(
     : ngraph::runtime::ExternalFunction(function, release_function)
     , m_compiled_function(nullptr)
     , m_emit_timing(std::getenv("NGRAPH_GPU_EMIT_TIMING") != nullptr)
+    , m_ctx(new GPURuntimeContext)
 {
+    // Create context use driver API and make it current, the runtime call will pickup the context
```

Comment:
why not put in `setup_runtime_context`? I'm think it's better to put cuda/cublas/cudnn initializations together?

Reply:
The problem is that the external_function::compile() is called prior to the consturction of the call_frame, and the gpu_emitters require the GPURuntimeContext. I'm open to suggestions on a cleaner place for this to be instantiated. We could possibly move the setup_runtime_context to GPUExternalFunction.